src/Infrastructure/Process/FileCopier.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+namespace PhpTuf\ComposerStager\Infrastructure\Process;
+
+use PhpTuf\ComposerStager\Exception\LogicException;
+use PhpTuf\ComposerStager\Exception\ProcessFailedException;
+use PhpTuf\ComposerStager\Infrastructure\Process\Runner\RsyncRunner;
+
+/**
+ * @internal
+ */
+class FileCopier
+{
+    /**
+     * @var \PhpTuf\ComposerStager\Infrastructure\Process\Runner\RsyncRunner
+     */
+    private $rsync;
+
+    public function __construct(RsyncRunner $rsync)
+    {
+        $this->rsync = $rsync;
+    }
+
+    /**
+     * @param string $from
+     * @param string $to
+     * @param string[] $exclusions Paths to exclude, relative to the "from" path.
+     * @param callable|null $callback An optional PHP callback to run whenever
+     *   there is some output available on STDOUT or STDERR.
+     *
+     * @see https://symfony.com/doc/current/components/process.html#running-processes-asynchronously
+     *
+     * @throws \PhpTuf\ComposerStager\Exception\ProcessFailedException
+     */
+    public function copy(string $from, string $to, array $exclusions, ?callable $callback = null): void
```

Comment:
Should `$exclusions` have a default value?

Reply:
Yeah, that makes sense. Done.